Forge.TreeWalker.UnitTests/test/TreeSchemaValidatorTests.cs

```diff
@@ -0,0 +1,215 @@
+        private string schemaDirectoryPath;
+        private string stringRules;
+        private JSchema jschemaRules;
+        private string invalidSchemaNotTree;
+        private string invalidSchemaDirectoryPath;
+        private string invalidSchemaWithErrorContent;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            schemaPath = Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas\\TardigradeSchema.json");
+            schemaDirectoryPath = Path.Combine(Environment.CurrentDirectory, "test\\ExampleSchemas");
+            stringRules = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "contracts\\ForgeSchemaValidationRules.json"));
+            jschemaRules = JSchema.Parse(stringRules);
+            stringSchema = File.ReadAllText(schemaPath);
+            stringSchemaList = "{ Tree1:" + stringSchema + ", Tree2:" + stringSchema + "}";
+            treeSchema =  JsonConvert.DeserializeObject<ForgeTree>((string)stringSchema);
+            treeSchemas = new Dictionary<string, ForgeTree>();
+            treeSchemas.Add("tree1", treeSchema);
+            treeSchemas.Add("tree2", treeSchema);
+            treeSchemas.Add("tree3", treeSchema);
+            invalidSchemaNotTree = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "test\\InvalidTestSchemas\\InvalidTestSchema.json"));
+            invalidSchemaWithErrorContent = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "test\\InvalidTestSchemas\\InvalidTestSchemaErrorContent.json"));
+            invalidSchemaDirectoryPath = "test\\ExampleSchemas\\TardigradeSchema.json";
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaInForgeTreeWithCustomRulesInString()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchema(treeSchema, stringRules, out List<string> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaInForgeTreeWithCustomRulesInJSchema()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchema(treeSchema, jschemaRules, out List<string> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaInForgeTreeListWithCustomRulesInString()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchema(treeSchema, stringRules, out List<string> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaInForgeTreeListWithCustomRulesInJSchema()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchema(treeSchema, jschemaRules, out List<string> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
```

Comment:
I think you meant to use treeSchemas in these 2 UTs. Update the method name as well (not dealing with list).